server.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const http = require('http');
+const SkiData = require('./model/ski-data.js');
+const Router = require('./lib/router.js');
+const data = require('./lib/data.js');
+const PORT = process.env.PORT || 8080;
+
+const routes = new Router();
+
+routes.get('/api/ski-data', (req, res) => {
+  if(req.url.query.id) {
+    data.getData('location', req.url.query.id)
+    .then( skiData => {
+      res.writeHead(200, {'Content-Type': 'application/json'});
+      res.write(JSON.stringify(skiData));
+      res.end();
+    })
+    .catch( err => {
+      console.error(err);
+      res.writeHead(404, {'Content-Type': 'text/plain'});
+      res.write('Not found');
+      res.end();
+    });
+    return;
+  }
+  if(!req.url.query.id) {
+    res.writeHead(400, {'Content-Type': 'text/plain'});
+    res.write('bad request');
+    res.end();
+  }
+});
+
+routes.post('/api/ski-data', (req, res) => {
+  try {
+    var skiData = new SkiData(req.body.location, req.body.rating);
+    data.setData('location', skiData);
+    res.writeHead(200, {'Content-Type': 'application/json'});
+    res.write(JSON.stringify(skiData));
+    res.end();
+  } catch(err) {
+    console.error(err);
+    res.writeHead(400, {'Content-Type': 'text/plain'});
+    res.write('bad request');
+    res.end();
+  }
+});
+
+// routes.put('/api/ski-data', (req, res) => {
+//TODO: add put route
+// });
```

Comment:
Glad to see you were thinking about PUT- minor nitpick- just be sure to remove commented out code before pushing to the githubs!